frontend/src/metabase/containers/ItemPicker/ItemPicker.tsx

```diff
@@ -100,6 +102,13 @@ function ItemPicker<TId>({
   const isPickingNotCollection = models.some(model => model !== "collection");
 
   const openCollection = collectionsById[openCollectionId];
+  const isOpenCollectionInPersonalCollection = openCollection?.is_personal;
+  const hideItems =
+    showOnlyPersonalCollections && !isOpenCollectionInPersonalCollection;
+
+  useEffect(() => {
```

Comment:
Maybe this is just me but I would prefer an if statement instead of the optional function call on this one

Reply:
Good point. This is worth discussing with FE team.

Reply:
I [asked the question here](https://metaboat.slack.com/archives/C505ZNNH4/p1698145752378499). Likely I'll keep the change as is. But the question already generated some useful insights. Thanks for asking!